driver/src/contracts/stablex_contract.rs

```diff
@@ -84,15 +82,28 @@ impl StableXContract for StableXContractImpl {
         let (prices, token_ids_for_price) = encode_prices_for_contract(solution.prices);
         let (owners, order_ids, volumes) =
             encode_execution_for_contract(orders, solution.executed_buy_amounts);
-
+        let improvement_denominator: U256 = self
+            .base
+            .contract
+            .query(
+                "IMPROVEMENT_DENOMINATOR",
+                (),
+                None,
+                Options::default(),
+                None,
+            )
+            .wait()?;
+        // In the smart contracts the utility value will be multiplied by 1 + IMPROVEMENT_DENOMINATOR
+        // hence, the following reduction of the maximal possible utility value is required:
+        let max_utility_value = U256::max_value() / (U256::one() + improvement_denominator);
```

Comment:
I'm wondering if we should just calculate this value once and hardcode it as a constant. Feels like it won't change enough to justify a dynamic call and computation on every run. What do you think?

Reply:
Maybe memoize it?

Reply:
Why not just hardcode as a constant?

Reply:
I am also in favor of hardcoding it for now. Just because we did it for other public constants for the smart contract as well. E.g. FeeDENOMINATOR.

But I am happy to change all of them to a memorized solution later, once we agreed on it.